### PR TITLE
3.0.0 Release

### PR DESCRIPTION
**Pull request by abonander**
The crate has been completely revamped with a new, much more Rust-y API.

### New Features
* Serde support (finally)
* Support for inline storage for hashes with generic arrays
* DCT and Difference-of-Gaussians preprocessing for all hash types
* Hash sizes are rectangular now to allow capturing more detail in one dimension or the other

Additionally, DCT is implemented by the [RustDCT](https://crates.io/crates/rustdct) crate now which enables the following:
* Precomputing of DCT coefficients is now automatic and per-instance of `HasherConfig` instead of user-configured per-thread
* DCT is implemented in terms of FFTs now and receives roughly an order of magnitude speedup (need to add benches back to show this)

### Breaking Changes
* The old API is completely gone
* Requirement for the `image` crate now starts at `0.21` (might be able to loosen this)
* Minimum supported Rust version is increased to at least 1.26 (need to establish this exactly)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,20 +14,26 @@ keywords = ["image", "hash", "perceptual", "difference"]
...
-default = ["rust-image"]
-rust-image = ["image"]
-bench = []
+nightly = []
 
 [dependencies]
-bit-vec = "0.4"
-rustc-serialize = "0.3"
+base64 = "0.10"
+image = "0.21"
```
Comment (abonander) [inline Cargo.toml]: `image 0.22` is out now, this should be set back to a range if possible.

**Comment by kpcyrd** [comment]
Any updates on this one?

**Comment by abonander** [comment]
I'll try to release it today when I get home. I was waiting for feedback and kinda forgot about it.

**Comment by abonander** [comment]
published `3.0.0`

**Pull request merged**